File: public/usage-examples/camera/move_camera_to/move_camera_to-1-example.cpp
#include "splashkit.h"

int main()
{
    open_window("Move Camera Demo", 800, 600);
    
    double target_x = 0, target_y = 0;
    
    while (!window_close_requested("Move Camera Demo"))
    {
        process_events();
        
        // Use arrow keys to move the camera
        if (key_down(RIGHT_KEY))
            target_x += 5;
        if (key_down(LEFT_KEY))
            target_x -= 5;
        if (key_down(DOWN_KEY))
            target_y += 5;
        if (key_down(UP_KEY))
            target_y -= 5;
        
        // Move camera to the new position
        move_camera_to(target_x, target_y);
        
        clear_screen(COLOR_DARK_GREEN);
        
        // Draw a large world with objects
        for (int x = -500; x < 1500; x += 100)
        {
            for (int y = -500; y < 1200; y += 100)
            {
                fill_circle(COLOR_RED, x, y, 20);
                draw_text(std::to_string(x) + "," + std::to_string(y), 
                         COLOR_WHITE, x - 15, y + 25);
            }
        }
        
        // Draw camera boundaries
        point_2d cam_pos = camera_position();
        draw_rectangle(COLOR_YELLOW, cam_pos.x, cam_pos.y, 800, 600);
        
        // Show instructions
        draw_text("Use arrow keys to move camera", COLOR_WHITE, 10, 10);
        draw_text("Camera at: (" + std::to_string((int)cam_pos.x) + ", " + 
                 std::to_string((int)cam_pos.y) + ")", COLOR_WHITE, 10, 30);
        
        refresh_screen(60);
    }
    
    close_window("Move Camera Demo");
    return 0;
}

File: public/usage-examples/camera/move_camera_to/move_camera_to-1-example.py
from splashkit import *

def main():
    open_window("Move Camera Demo", 800, 600)
    
    target_x, target_y = 0, 0
    
    while not window_close_requested("Move Camera Demo"):
        process_events()
        
        # Use arrow keys to move the camera
        if key_down(key_code.right_key):
            target_x += 5
        if key_down(key_code.left_key):
            target_x -= 5
        if key_down(key_code.down_key):
            target_y += 5
        if key_down(key_code.up_key):
            target_y -= 5
        
        # Move camera to the new position
        move_camera_to(target_x, target_y)
        
        clear_screen(color_dark_green())
        
        # Draw a large world with objects
        for x in range(-500, 1500, 100):
            for y in range(-500, 1200, 100):
                fill_circle(color_red(), x, y, 20)
                draw_text(f"{x},{y}", color_white(), x - 15, y + 25)
        
        # Draw camera boundaries
        cam_pos = camera_position()
        draw_rectangle(color_yellow(), cam_pos.x, cam_pos.y, 800, 600)
        
        # Show instructions
        draw_text("Use arrow keys to move camera", color_white(), 10, 10)
        draw_text(f"Camera at: ({int(cam_pos.x)}, {int(cam_pos.y)})", 
                  color_white(), 10, 30)
        
        refresh_screen(60)
    
    close_window("Move Camera Demo")

if __name__ == "__main__":
    main()
